pkg/sdn/registry/netnamespace/rest.go

```diff
@@ -0,0 +1,163 @@
+package netnamespace
+
+/*
+Only needed for backward compatibility with older clients (openshift <= 3.2 release)
+*/
+
+import (
+	"fmt"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrs "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/rest"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	"github.com/openshift/openshift-sdn/pkg/netid"
+	"github.com/openshift/origin/pkg/sdn/api"
+)
+
+// REST implements the RESTStorage interface backed by namespace and it only supports the adding or deleting VNID annotation on the namespace.
+type REST struct {
+	client kclient.NamespaceInterface
+}
+
+// NewREST returns a new REST.
+func NewREST(client kclient.NamespaceInterface) *REST {
+	return &REST{client: client}
+}
+
+// New returns a new NetNamespace
+func (r *REST) New() runtime.Object {
+	return &api.NetNamespace{}
+}
+
+// NewList returns a new NetNamespaceList
+func (*REST) NewList() runtime.Object {
+	return &api.NetNamespaceList{}
+}
+
+var _ = rest.Getter(&REST{})
+
+func (s *REST) Get(ctx kapi.Context, name string) (runtime.Object, error) {
+	return s.getNetNamespace(name)
+}
+
+var _ = rest.Creater(&REST{})
+
+func (s *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, error) {
+	netns, ok := obj.(*api.NetNamespace)
+	if !ok {
+		return nil, kerrs.NewBadRequest("invalid type")
+	}
+	Strategy.PrepareForCreate(netns)
+	return s.update(ctx, netns)
+}
+
+var _ = rest.Updater(&REST{})
+
+func (s *REST) Update(ctx kapi.Context, obj runtime.Object) (runtime.Object, bool, error) {
+	netns, ok := obj.(*api.NetNamespace)
+	if !ok {
+		return nil, false, kerrs.NewBadRequest("invalid type")
+	}
+	Strategy.PrepareForUpdate(netns, nil)
+	updatedNetNs, err := s.update(ctx, netns)
+	return updatedNetNs, false, err
```

Comment:
in theory, you should handle the "created" flag (either by refusing to update the netns if it doesn't already exist, or by returning `true` rather than `false` if you created it)